src/services/accession.jl

```diff
@@ -0,0 +1,217 @@
+end
+
+==(x::GenBank, y::GenBank) = x.version == y.version && x.accession == y.accession
+
+function parse(::Type{Accession{:GenBank}}, s::String)
+    s = strip(s)
+    dot = search(s, '.')
+    @assert dot > 0
+    accession = s[1:dot-1]
+    version = parse(Uint8, s[dot+1:end])
+    return Accession{:GenBank,GenBank}(GenBank(accession, version))
+end
+
+function show(io::IO, genbank::Accession{:GenBank})
+    print(io, genbank.data.accession)
+    print(io, '.')
+    print(io, genbank.data.version)
+end
+
+hash(genbank::GenBank) = hash(genbank.accession) $ hash(genbank.version)
+
+@osx_only function browse(genbank::Accession{:GenBank})
+    run(`open http://www.ncbi.nlm.nih.gov/nuccore/$genbank`)
+end
+
+
+# RefSeq
+
+immutable RefSeq
+    prefix::Uint8
+    number::Uint32
+    version::Uint8
+end
+
+# generate lookup table
+macro gen_table(name, typ, xs)
+    xs = eval(xs)
+    enc_table = symbol(string(name, :_encode))
+    dec_table = symbol(string(name, :_decode))
+    esc(quote
+        const $enc_table = [x => convert($typ, i) for (i, x) in enumerate($xs)]
+        const $dec_table = $xs
+    end)
+end
+
+# http://www.ncbi.nlm.nih.gov/books/NBK21091/ (Table 1)
+@gen_table refseq_prefix Uint8 [
+    "AC_", "NC_", "NG_", "NT_", "NW_",
+    "NS_", "NZ_", "NM_", "NR_", "XM_",
+    "XR_", "AP_", "NP_", "YP_", "XP_",
+    "ZP_",
+]
+
+function ismatch(::Type{Accession{:RefSeq}}, s::String)
+    ismatch(r"^\s*(:?A[CP]|N[CGTWSZMRP]|X[MRP]|YP|ZP)_\d{6}(:?\d{3})?\.\d+\s*$", s)
+    #             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~   ^~
+    #             prefix                              number            version
+end
+
+function parse(::Type{Accession{:RefSeq}}, s::String)
+    s = strip(s)
+    prefix = refseq_prefix_encode[s[1:3]]
+    dot = search(s, '.')
+    @assert dot > 0
+    number = parse(Uint32, s[4:dot-1])
+    version = parse(Uint8, s[dot+1:end])
+    return Accession{:RefSeq,RefSeq}(RefSeq(prefix, number, version))
+end
+
+function show(io::IO, refseq::Accession{:RefSeq})
+    prefix = refseq_prefix_decode[refseq.data.prefix]
+    nd = ndigits(refseq.data.number)
+    if nd <= 6
+        @printf io "%s%06d.%d" prefix refseq.data.number refseq.data.version
+    elseif nd <= 9
+        @printf io "%s%09d.%d" prefix refseq.data.number refseq.data.version
+    else
+        error("number overflow")
+    end
+end
+
+hash(refseq::RefSeq) = hash(refseq.prefix) $ hash(refseq.number) $ hash(refseq.version)
+
+@osx_only function browse(refseq::Accession{:RefSeq})
+    run(`open http://www.ncbi.nlm.nih.gov/nuccore/$refseq`)
+end
+
+# Consensus CDS (CCDS)
+
+# http://www.ncbi.nlm.nih.gov/CCDS/CcdsBrowse.cgi#ccdsIds
+
+immutable CCDS
+    number::Uint32
+    version::Uint8
+end
+
+function ismatch(::Type{Accession{:CCDS}}, s::String)
+    return ismatch(r"^\s*CCDS\d+\.\d+\s*$", s)
+end
+
+
+# Ensembl
+
+# http://www.ensembl.org/info/genome/stable_ids/index.html
+
+function ismatch(::Type{Accession{:Ensembl}}, s::String)
```

Comment:
Since Ensembl is versioned, and the gene, protein, transcript, etc, referred to may change between versions, it seems important to associate the Ensembl version with the identifier. I'm not sure how best to handle that.